io/io/src/RRawFileUnix.cxx

```diff
@@ -115,26 +115,33 @@ void ROOT::Internal::RRawFileUnix::OpenImpl()
 void ROOT::Internal::RRawFileUnix::ReadVImpl(RIOVec *ioVec, unsigned int nReq)
 {
 #ifdef R__HAS_URING
-   if (RIoUring::IsAvailable()) {
-      RIoUring ring;
-      std::vector<RIoUring::RReadEvent> reads;
-      reads.reserve(nReq);
-      for (std::size_t i = 0; i < nReq; ++i) {
-         RIoUring::RReadEvent ev;
-         ev.fBuffer = ioVec[i].fBuffer;
-         ev.fOffset = ioVec[i].fOffset;
-         ev.fSize = ioVec[i].fSize;
-         ev.fFileDes = fFileDes;
-         reads.push_back(ev);
+   thread_local bool uring_failed = false;
```

Comment:
It is not clear whether this is faster than
```
static std::atomic<bool> uring_failed = false;
```
(the thread local has to lookup what is the local storage for each time through).